src/components/PopupWithForm.js

```diff
@@ -1,20 +1,41 @@
-import Popup from './Popup.js';
+import Popup from "./Popup.js";
 
-class PopupWithForm extends Popup {
-    constructor(popupSelector, handleFormSubmit) {
-        super({ popupSelector});
-        this.popupForm = this._popupElement.querySelector('.popup__form');
-        this._handleFormSubmit = handleFormSubmit;
+export default class PopupWithForm extends Popup {
+  constructor(popupSelector, submitCallback) {
+    super(popupSelector);
+    this._submitCallback = submitCallback;
+    this._form = this._popup.querySelector("form");
+    this._inputs = this._form.querySelectorAll("input");
+    this._submitButton = this._form.querySelector(".modal__button");
+    this._defaultButtonText = this._submitButton.textContent;
+  }
 
```

Comment:
**Could be improved**

You can make ` setInputValues` in class `PopupWithForm`, which can insert data into inputs:

```js

  setInputValues(data) {
    this._inputList.forEach((input) => {
      // Here you insert the `value` by the `name` of the input
      input.value = data[input.name];
    });
  }



```

So, this way you’ll not have to search the inputs of the profile in `index.js`